ext/ffi/00_ffi.js

```diff
@@ -142,6 +142,59 @@
     }
   }
 
+  function prepareParameters(args, types) {
+    const parameters = [];
+    const buffers = [];
+    let fn = null;
+
+    for (let i = 0; i < types.length; i++) {
+      const type = types[i];
+      const arg = args[i];
+
+      if (typeof type === "object" && "function" in type) {
+        if (typeof arg !== "function") {
+          throw new TypeError(
+            `Expected function got ${typeof arg}`,
+          );
+        }
+
+        parameters.push(i);
+        const fnTypes = type.function.parameters;
+        fn = (...args) => {
+          const ptrIndices = fnTypes.map((ty, i) => ty == "pointer" ? i : -1)
```

Comment:
Some optimization:
```ts
fn = (...args) => {
  for (let i = 0; i < args.length; i++) {
    const ty = args[i];
    if (ty === "pointer") {
      args[i] = new UnsafePointer(unpackU64[ty]);
    }
  }
  return arg(...args);
}
```

Reply:
This won't work. `args[i]` is the actual value and not the type.

Reply:
Updated:
```javascript
fn = (...args) => {
  for (let i = 0; i < fnTypes.length; i++) {
    const ty = fnTypes[i];
    if (ty === "pointer") {
      args[i] = new UnsafePointer(unpackU64(args[i]));
    }
  }
  return arg(...args);
}
```